tests/lib/rules/arrow-body-style.js

```diff
@@ -45,6 +45,77 @@ ruleTester.run("arrow-body-style", rule, {
         { code: "var foo = () => { return { bar: 0 }; };", options: ["as-needed", { requireReturnForObjectLiteral: true }] }
     ],
     invalid: [
+        {
+            code: "for (let a = (b, c, d) => { return vb && c in d; }; ;);",
+            output: "for (let a = (b, c, d) => (vb && c in d); ;);",
+            errors: [
+                {
+                    line: 1,
+                    column: 27,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "for (let a = (b, c, d) => { return v in b && c in d; }; ;);",
+            output: "for (let a = (b, c, d) => (v in b && c in d); ;);",
+            errors: [
+                {
+                    line: 1,
+                    column: 27,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "for (let a = (b, c, d) => { return vb in dd ; }; ;);",
+            output: "for (let a = (b, c, d) => (vb in dd ); ;);",
+            errors: [
+                {
+                    line: 1,
+                    column: 27,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "for (let a = (b, c, d) => { return vb in c in dd ; }; ;);",
+            output: "for (let a = (b, c, d) => (vb in c in dd ); ;);",
+            errors: [
+                {
+                    line: 1,
+                    column: 27,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "do{let a = () => {return f in ff}}while(true){}",
+            output: "do{let a = () => (f in ff)}while(true){}",
+            errors: [{
+                line: 1,
+                column: 18,
+                messageId: "unexpectedSingleBlock"
+            }]
+        },
+        {
+            code: "scores.map(score => { return x in +(score / maxScore).toFixed(2)});",
+            output: "scores.map(score => (x in +(score / maxScore).toFixed(2)));",
+            errors: [{
+                line: 1,
+                column: 21,
+                messageId: "unexpectedSingleBlock"
+            }]
+        },
+        {
+            code: "const fn = (a, b) => { return a + x in Number(b) };",
+            output: "const fn = (a, b) => (a + x in Number(b));",
+            errors: [{
+                line: 1,
+                column: 22,
+                messageId: "unexpectedSingleBlock"
+            }]
+        },
```

Comment:
I added wrap if `in` operator is found just to be safe if there is any kind of issue. It will add wrap even if it's valid just to minimize the complexity otherwise we need to validate the code (I am not sure how to do so for a particular block).

Reply:
@anikethsaha 
> I added wrap if in operator is found just to be safe if there is any kind of issue.

IMO, the `arrow-body-style` fixer should not add extra parentheses. (except for within "for", because it's not extra parentheses)

+ Adding extra parenthesis in the `arrow-body-style` can make conflict with `no-extra-parens`.

```js
/* eslint no-extra-parens: ["error", "all"] */

for (let a = (b, c, d) => (c in d); ;); // No Error, because these are not extra parentheses 
 
do{
  let a = () => (f in ff) // Error
}while(true){}

scores.map(score => (x in +(score / maxScore).toFixed(2))); // Error
 
const fn = (a, b) => (a + x in Number(b)); // Error
```

Reply:
@yeonjuan can you check the latest commit, if it's fine by you